File: app/src/main/java/me/texy/treeview/MainActivity.java
package me.texy.treeview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import me.texy.treeview.treeview.TreeNode;
import me.texy.treeview.treeview.TreeView;

public class MainActivity extends AppCompatActivity {

    private ViewGroup viewGroup;
    private TreeNode root;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        root = TreeNode.root();
        buildTree();
        viewGroup = (RelativeLayout) findViewById(R.id.container);

        TreeView treeView = new TreeView(root, this);
        treeView.setBaseNodeViewFactory(new MyNodeViewFactory());
        viewGroup.addView(treeView.getView());
    }

    private void buildTree() {
        for (int i = 0; i < 20; i++) {
            TreeNode treeNode = new TreeNode(new String("Parent  " + "No." + i*100));
            treeNode.setLevel(0);
            for (int j = 0; j < 5; j++) {
                TreeNode treeNode1 = new TreeNode(new String("Child " + "No." + i*100+10*j));
                treeNode1.setLevel(1);
                for (int k = 0; k < 5; k++) {
                    TreeNode treeNode2 = new TreeNode(new String("Grand Child " + "No." + i*100+j*10+k));
                    treeNode2.setLevel(2);
                    treeNode1.addChild(treeNode2);
                }
                treeNode.addChild(treeNode1);
            }
            root.addChild(treeNode);
        }
    }

}

File: app/src/main/java/me/texy/treeviewdemo/MainActivity.kt
package me.texy.treeviewdemo

import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import dagger.hilt.android.AndroidEntryPoint
import me.texy.treeviewdemo.databinding.ActivityMainBinding
import me.texy.treeviewdemo.ui.screen.hone.CompetitionFragment

@AndroidEntryPoint
open class MainActivity : AppCompatActivity(),
    BottomNavigationView.OnNavigationItemSelectedListener {
    private lateinit var activityMainBinding: ActivityMainBinding
    var member: Int? = null
    val value by lazy {
        member = 6
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
//        test(value)
//        val navHostFragment =
//            supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
//        val navController = navHostFragment.navController
//        findViewById<BottomNavigationView>(R.id.bottom_nav)
//            .setupWithNavController(navController)
        supportFragmentManager.beginTransaction()
            .add(R.id.nav_host_fragment, CompetitionFragment())
            .commit()
        activityMainBinding.bottomNav.setOnNavigationItemSelectedListener(this)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.home_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return super.onOptionsItemSelected(item)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.competitions -> {
                val competitionFragment = CompetitionFragment()
                show(competitionFragment)
                return true
            }
        }
        return false
    }

    private fun show(fragment: Fragment) {

        val fragmentManager = supportFragmentManager

        fragmentManager
            .beginTransaction()
            .replace(R.id.nav_host_fragment, fragment)
            .commit()
    }

    fun test(value: Any) {
        Log.d("TAG999", "test:  " + member)
    }
}